cryptarchia/cryptarchia.py

```diff
@@ -118,37 +122,103 @@ def on_block(self, block: BlockHeader):
             self.forks.append(new_chain)
         # otherwise, we're missing the parent block
         # in that case, just ignore the block
-        
 
     # Evaluate the fork choice rule and return the block header of the block that should be the head of the chain
     def fork_choice(local_chain: Chain, forks: List[Chain]) -> BlockHeader:
         pass
-        
 
     def tip(self) -> BlockHeader:
         return self.fork_choice()
 
 
 @dataclass
 class Coin:
+    pk: int
     value: int
 
+
+@dataclass
+class LedgerState:
+    """
+    A snapshot of the ledger state up to some height
+    """
+
+    height: int = None
```

Comment:
height is not a unique identifier of ledger state

Reply:
true, perhaps a block id instead.